mdoc/Mono.Documentation/Updater/Formatters/JsUsageFormatter.cs

```diff
@@ -0,0 +1,246 @@
+                // If you want to pass a Windows Runtime structure to a Windows Runtime method, 
+                // don't instantiate the structure with the new keyword. Instead, create an object 
+                // and add the relevant members and their values. The names of the members should be in camel case:
+                // SomeStruct.firstMember.
+
+                // var [struct name, camel cased] = {
+                //	[fieldname, came cased] : /* Your value */
+                buf.Append("var ");
+                buf.Append(CamelCase(GetName(type)));
+                buf.Append(" = {");
+                buf.Append(GetLineEnding());
+                buf.Append(string.Join("," + GetLineEnding(), 
+                    type.Fields.Select(i => CamelCase(i.Name) + " : /* Your value */")));
+                buf.Append(GetLineEnding());
+                buf.Append("}");
+                return buf.ToString();
+            }
+            if (DocUtils.IsDelegate(type))
+            {
+                //  var [delegateName, camel-cased]Handler = function([parameter name list, camel cased]){
+                //  /* Your code */
+                //            }
+                MethodDefinition invoke = type.GetMethod("Invoke");
+                buf.Append("var ");
+                buf.Append(CamelCase(GetName(type)));
+                buf.Append("Handler = function(");
+                AppendParameters(buf, invoke, invoke.Parameters);
+                buf.Append("){");
+                buf.Append(GetLineEnding());
+                buf.Append("/* Your code */");
+                buf.Append(GetLineEnding());
+                buf.Append("}");
+                return buf.ToString();
+            }
+            
+            var publicConstructor = GetConstructor(type);
+            return GetDeclaration(publicConstructor);
+        }
+
+        protected override string GetMethodDeclaration(MethodDefinition method)
+        {
+            var buf = new StringBuilder();
+
+            if (IsAsync(method))
+            {
+                // Async Methods
+                // (For static methods, use the fully-qualified class name. For non-static, 
+                // use the class name without qualification and camel-cased.)
+                // [fully qualified type name].[camel-cased method name]([parameter names]).done( /* Your success and error handlers */ );
+                var typeName = CamelCase(GetName(method.DeclaringType));
+                if (method.IsStatic)
+                    typeName = ProcessFullName(method.DeclaringType.FullName);
+
+                buf.Append(typeName);
+                buf.Append(".");
+                buf.Append(GetMethodName(method));
+                buf.Append("(");
+                AppendParameters(buf, method, method.Parameters);
+                buf.Append(").done( /* Your success and error handlers */ )");
+
+                return buf.ToString();
+            }
+
+            // Non-static
```

Comment:
This comment is ambiguous ... it looks like it was copied in here from documentation as reference? Can you clarify in clearer prose this comment? It's not immediately clear whether this is relevant to the code before or after it.

Reply:
Done